elfpy/pricing_models/hyperdrive.py

```diff
@@ -257,39 +262,48 @@ def calc_out_given_in(
         TradeResult
             The result of performing the trade.
         """
+
+        # pylint: disable=too-many-locals
+
         # Calculate some common values up front
         in_amount = Decimal(in_.amount)
-        normalized_time = Decimal(time_remaining.normalized_time)
+        curve_portion = Decimal(time_remaining.normalized_time)
+        flat_portion = 1 - curve_portion
         share_price = Decimal(market_state.share_price)
-        d_bonds = in_amount * (1 - normalized_time)
+        d_bonds = in_amount * flat_portion  # whether in_.unit is base or pt, at maturity d_bonds = d_base
         d_shares = d_bonds / share_price
         # Redeem the matured bonds 1:1 and simulate these updates hitting the reserves.
-        market_state = market_state.copy()  # don't want to modify the actual market state
+        market_state_simulated = market_state.copy()  # don't want to modify the actual market state
```

Comment:
`market_state_simulated = market_state.copy()` causes us to store `market_state` twice in memory, while the other version ` market_state = market_state.copy() ` does not. If we don't _need_ two versions, then let's leave it how it was.